src/installer/__main__.py

```diff
@@ -39,16 +46,27 @@ def main_parser() -> argparse.ArgumentParser:
     return parser
 
 
-def get_scheme_dict(distribution_name: str) -> Dict[str, str]:
+def get_scheme_dict(
+    distribution_name: str, prefix: Optional[str] = None
+) -> Dict[str, str]:
     """Calculate the scheme dictionary for the current Python environment."""
-    scheme_dict = sysconfig.get_paths()
+    vars = {}
+    if prefix is not None:
+        vars["base"] = vars["platbase"] = prefix
+
+    scheme_dict = sysconfig.get_paths(vars=vars)
 
     # calculate 'headers' path, not currently in sysconfig - see
     # https://bugs.python.org/issue44445. This is based on what distutils does.
     # TODO: figure out original vs normalised distribution names
     scheme_dict["headers"] = os.path.join(
         sysconfig.get_path(
-            "include", vars={"installed_base": sysconfig.get_config_var("base")}
+            "include",
+            vars={
+                "installed_base": prefix
+                if prefix is not None
+                else sysconfig.get_config_var("base")
```

Comment:
Instead of having a conditional inline on an already deeply nested block of code, could you move this out to a local that's assigned in two branches of an if-else statement?

That way the line coverage will actually require covering both possible branches here in the tests.

Reply:
Done!